examples/ComplexDataTypes/README.md

```diff
@@ -0,0 +1,217 @@
+# Complex Data Types Example
+
+This example demonstrates how to use complex data types (structs) with iceoryx2 in
+C#. It shows how to define custom structs that can be sent and received across
+process boundaries using zero-copy shared memory communication.
+
+## Features
+
+This example demonstrates:
+
+1. **Simple Structs**: `TransmissionData` with primitive fields
+2. **Sensor Data**: More realistic struct with timestamps and sensor readings
+3. **Fixed Arrays**: `Point3D` with embedded fixed-size arrays
+4. **Cross-Language Compatibility**: Using `[Iox2Type]` attribute for type name mapping
+5. **Memory Layout Control**: Using `[StructLayout(LayoutKind.Sequential)]`
+   for C compatibility
+
+## Key Concepts
+
+### Memory Layout
+
+For structs to work correctly across language boundaries, they must have a
+predictable memory layout:
+
+```csharp
+[StructLayout(LayoutKind.Sequential)]  // Ensures fields are laid out in order
+public struct TransmissionData
+{
+    public int X;      // 4 bytes
+    public int Y;      // 4 bytes
+    public double Funky;  // 8 bytes
+    // Total: 16 bytes (plus any padding)
+}
+```
+
+### Type Name Mapping
+
+The `[Iox2Type]` attribute allows you to specify the type name used for cross-
+language communication:
+
+```csharp
+[Iox2Type("TransmissionData")]  // Must match the name used in Rust/C
+public struct TransmissionData
+{
+    // ...
+}
+```
+
+Without this attribute, the C# type name (e.g., "TransmissionData") is used by default.
+
+### Unmanaged Constraint
+
+All types used with iceoryx2 must be `unmanaged`, meaning they:
+
+* Cannot contain reference types (classes, strings, etc.)
+* Cannot contain managed pointers
+* Can only contain other unmanaged types
+* Can be safely copied byte-by-byte
+
+```csharp
+// ✓ Valid unmanaged types
+public struct SensorData
+{
+    public long Timestamp;
+    public float Temperature;
+    public float Humidity;
+    public int SensorId;
+}
+
+// ✗ Invalid - contains managed type (string)
+public struct InvalidData
+{
+    public int Id;
+    public string Name;  // ERROR: string is a managed type
+}
+```
+
+## How to Build
+
+```bash
+cd /path/to/iceoryx2/iceoryx2-ffi/csharp
+dotnet build examples/ComplexDataTypes/ComplexDataTypes.csproj
+```
+
+## How to Run
+
+### Terminal 1 - Publisher
+
+```bash
+cd examples/ComplexDataTypes
+dotnet run -- publisher transmission
```

Comment:
I think the iceoryx2 types have changed in between, can be something like SensorData. Does apply for the other commands here as well